src/QsCompiler/DocumentationGenerator/ProcessDocComments.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Quantum.QsCompiler;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.Documentation;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.Core;
+using Newtonsoft.Json.Linq;
+
+using Range = Microsoft.Quantum.QsCompiler.DataTypes.Range;
+
+#nullable enable
+
+namespace Microsoft.Quantum.Documentation
+{
+    public class ProcessDocComments
+    : SyntaxTreeTransformation<ProcessDocComments.TransformationState>
+    {
+        public class TransformationState
+        { }
+
+        private readonly DocumentationWriter? writer;
+
+        public ProcessDocComments(
+            string? outputPath = null,
+            string? packageName = null
+        )
+        : base(new TransformationState())
+        {
+            writer = outputPath == null
+                     ? null
+                     : new DocumentationWriter(outputPath, packageName);
+
+            // We provide our own custom namespace transformation, and expression kind transformation.
+            this.Namespaces = new ProcessDocComments.NamespaceTransformation(this, writer);
+        }
+
+        public QsCompilation OnCompilation(QsCompilation compilation) =>
```

Comment:
This can be removed; OnCompilation is now defined as part of SyntaxTreeTransformation.